app/script/view_model/content/PreferencesOptionsViewModel.js

```diff
@@ -71,6 +71,19 @@ z.viewModel.content.PreferencesOptionsViewModel = class PreferencesOptionsViewMo
     amplify.publish(z.event.WebApp.CONNECT.IMPORT_CONTACTS, z.connect.ConnectSource.ICLOUD);
   }
 
+  save_call_logs() {
+    if (z.util.CallingLog.length < 1) {
```

Comment:
```js
if (!z.util.CallingLog.length) {
```

Reply:
```
const b = Object.create([]);
b.length = -1;
if (!b.length) {
console.log('Empty');
} else {
console.log('Full of stuff');
}
```

Reply:
True but modifying `b.length = 4` will also going to cause the same issues and that's why we should not modify `length` :)